.github/workflows/build-and-test.yml

```diff
@@ -289,15 +289,15 @@ jobs:
         if: startsWith( matrix.go-version, '1.23' ) != true
         run: make gotest GROUP=${{ matrix.group }}
       - name: Run Unit Tests With JUnit and Coverage
-        if: startsWith( matrix.go-version, '1.23' ) # only run junit/coverage on one version
+        if: startsWith( matrix.go-version, '1.23' ) != true # only run junit/coverage on one version
```

Comment:
I think this was missed out in the previous PRs (Go1.22 -> Go1.23)... `startsWith( matrix.go-version, '1.23' )` is correct, `startsWith( matrix.go-version, '1.23' ) != true` is wrong, so instead you would want to change the latter to the former everywhere else

cc @mx-psi can correct me

Reply:
I think Yang is right here, yep

Reply:
I feel like there's something we're still missing 🤔 

`Run Unit Tests`, as it's configured today, should only run when the go version doesn't start with 1.23. Therefore it should only run for go 1.24. This is not the behavior I'm seeing.:
* It runs for 1.23[[1](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/13524013972/job/37790853058)]
* It runs for 1.24[[2](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/13524013972/job/37790850656)]


Now for the JUnit tests, it is configured to run when it starts with 1.23, but it's not running for any of the go versions.


My hypothesis is that `startsWith(...)` is not returning anything, and therefore `!= true` is triggering the unit tests because empty is different than true